org.eclipse.winery.topologymodeler.ui/src/app/canvas/canvas.component.ts

```diff
@@ -0,0 +1,1403 @@
+        node.connectorEndpointVisible = !node.connectorEndpointVisible;
+        if (node.connectorEndpointVisible === true) {
+            this.dragSourceActive = false;
+            this.resetDragSource(nodeId);
+            for (const currentNode of this.nodeChildrenArray) {
+                if (currentNode.nodeAttributes.id !== nodeId) {
+                    if (currentNode.connectorEndpointVisible === true) {
+                        currentNode.connectorEndpointVisible = false;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Sets drag source which marks the area where a connection can be dragged from and binds to the connections listener
+     * @param dragSourceInfo
+     */
+    setDragSource(dragSourceInfo: any): void {
+        if (!this.dragSourceActive) {
+            this.newJsPlumbInstance.makeSource(dragSourceInfo.dragSource, {
+                connectorOverlays: [
+                    ['Arrow', {location: 1}],
+                ],
+            });
+            this.dragSourceInfos = dragSourceInfo;
+            this.newJsPlumbInstance.makeTarget(this.allNodesIds);
+            this.dragSourceActive = true;
+            this.bindConnection();
+        }
+    }
+
+    /**
+     * Handler for the DEL-Key - removes a node and resets everything associated with that deleted node
+     * @param event Keyboard event.
+     */
+    @HostListener('document:keydown.delete', ['$event'])
+    handleDeleteKeyEvent(event: KeyboardEvent) {
+        this.unbindConnection();
+        // if name, min or max instances has changed, do not delete the node.
+        if (this.selectedNodes.length > 0) {
+            let selectedNodeSideBarVisible = false;
+            for (const node of this.nodeChildrenArray) {
+                if (node.makeSelectionVisible === true) {
+                    if (!selectedNodeSideBarVisible) {
+                        this.hideSidebar();
+                    }
+                    selectedNodeSideBarVisible = true;
+                    this.newJsPlumbInstance.deleteConnectionsForElement(node.nodeAttributes.id);
+                    this.newJsPlumbInstance.removeAllEndpoints(node.nodeAttributes.id);
+                    this.newJsPlumbInstance.removeFromAllPosses(node.nodeAttributes.id);
+                    if (node.connectorEndpointVisible === true) {
+                        if (this.newJsPlumbInstance.isSource(node.dragSource)) {
+                            this.newJsPlumbInstance.unmakeSource(node.dragSource);
+                        }
+                    }
+                    this.ngRedux.dispatch(this.actions.deleteNodeTemplate(node.nodeAttributes.id));
+                }
+            }
+            this.selectedNodes.length = 0;
+        }
+
+    }
+
+    /**
+     * Removes the selected Nodes from JSPlumb and internally.
+     */
+    clearSelectedNodes(): void {
+        if (this.selectedNodes.length > 0) {
+            for (const node of this.nodeChildrenArray) {
```

Comment:
nodeChildrenArray is a @ViewChildren instance and has all node templates as childs. therefore only because of it one can access the node attribute "makeSelectionVisible"